src/main/java/project/service/user/UserServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package project.service.user;
+
+import jakarta.persistence.EntityNotFoundException;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import project.dto.user.CreateUserDto;
+import project.dto.user.UserDto;
+import project.exeption.RegistrationException;
+import project.mapper.UserMapper;
+import project.model.Role;
+import project.model.User;
+import project.repository.role.RoleRepository;
+import project.repository.user.UserRepository;
+
+@Service
```

Comment:
add `@Transactional`above the services
```suggestion
@Service
@Transactional
```

Reply:
add `@Transactional` for services